analysis-module/src/analysis_cproc.py

```diff
@@ -0,0 +1,146 @@
+# Python Imports
+import re
+import sys
+import json
+from datetime import datetime
+
+# Project Imports
+from src import vars
+from src import analysis_logger
+
+def extract_node_id(s: str) -> str:
+    pattern = r"node-(\d+)\.toml"
+    match = re.search(pattern, s)
+    if match:
+        return f"node_{match.group(1)}"
+    else:
+        return None
+
+
+def load_process_level_metrics(metrics_file_path: str):
+    
+    metrics_dict = {}
+    
+    try:
+        with open(metrics_file_path, 'r') as file:
+            
+            metrics_obj = json.load(file)
+            
+            info = metrics_obj['header']
+            all_samples = metrics_obj['containers']
+            nodes_cnt = 0
+
+            if len(all_samples) != info['num_containers']:
+                G_LOGGER.error('Number of containers in header does not match number of containers in samples')
+                return {}, None
+
+            for container_id, container_data in all_samples.items():
+
+                # tomls file names are unique per node
+                container_nodes = {}
+                for process in container_data['info']['processes']:
+                    
+                    node_id = extract_node_id(process['binary'])
+                    if not node_id:
+                        G_LOGGER.error('Couldn\'t match %s to node id in container %s' %(process['binary'], container_id))
+                        continue
+                    
+                    pid = process['pid']
+                    container_nodes[pid] = node_id  
+                
+                # Parse samples for each node
+                for sample in container_data['samples']:
+                    
+                    if sample['PID'] not in container_nodes:
+                        G_LOGGER.error('Couldn\'t find node id for PID %d in container %s' %(sample['PID'], container_id))
+                        continue
+                    
+                    node_id = container_nodes[sample['PID']]
+                    if not node_id:
+                        G_LOGGER.error('Couldn\'t find node id for PID %d in container %s' %(sample['PID'], container_id))
+                        continue
+
+                    if node_id in metrics_dict:
+                        metrics_dict[node_id]['samples'].append(sample)
+                    else:
+                        nodes_cnt += 1
+                        metrics_dict[node_id] = {'samples' : [sample]}
+            
+    except Exception as e:
+        analysis_logger.G_LOGGER.error('%s: %s' % (e.__doc__, e))
+        sys.exit()
+
+    analysis_logger.G_LOGGER.info('Loaded metrics for %d nodes.' %len(metrics_dict))
+
+    # for node_id, node_data in metrics_dict.items():
+    #     G_LOGGER.info('Node %s has %d samples' %(node_id, len(node_data['samples'])))
+
+    return metrics_dict, info
```

Comment:
I think that this function is too big. In order to add tests, I recommend you to split it in several smaller functions, or a propper testing will be basically almost impossible to do.